### ISSUE TITLE
DnsEndPointTest.Socket_StaticConnectAsync_HostNotFound failed in CI on Linux

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#3460]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/2733/console
```
13:39:16    System.Net.Sockets.Tests.DnsEndPointTest.Socket_StaticConnectAsync_HostNotFound [FAIL]
13:39:16       Assert.True() Failure
13:39:16       Expected: True
13:39:16       Actual:   False
13:39:16       Stack Trace:
13:39:17             at System.Net.Sockets.Tests.DnsEndPointTest.Socket_StaticConnectAsync_HostNotFound()
```

**Comment by pgavlin** [comment]
This is this assert on line 75: https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets/tests/FunctionalTests/DnsEndPointTest.cs#L75

This assert is firing because the connect is failing synchronously, which may actually be valid behavior. I'll look into it.

**Comment by stephentoub** [comment]
This also just failed on Windows:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_release/1337/

cc: @davidsh, @cipop

**Comment by pgavlin** [comment]
Closed by #3541.
